app/src/main/java/org/coepi/android/cen/Utils.kt

```diff
@@ -0,0 +1,103 @@
+package org.coepi.android.cen
+
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.security.MessageDigest
+import java.util.*
+
+@ExperimentalUnsignedTypes
+fun compareByteArray(a : UByteArray, b : UByteArray) : Int {
+    for (i in 0..a.size) {
+        if (a[i] != b[i]) {
+            return a[i].toInt() - b[i].toInt()
+        }
+    }
+    return 0
+}
+
+@ExperimentalUnsignedTypes
+fun computeUUIDHash(selfUUID: UUID, otherUUID: UUID) : ByteArray {
+    val md = MessageDigest.getInstance("SHA-256")
+    val selfBytes = getBytesFromUUID(selfUUID).toUByteArray()
+    val otherBytes = getBytesFromUUID(otherUUID).toUByteArray()
+    var bytes: UByteArray
+    if ( compareByteArray(selfBytes, otherBytes) < 0 ) {
+        bytes = selfBytes + otherBytes
+    } else {
+        bytes = otherBytes + selfBytes
+    }
+    return md.digest(bytes.toByteArray()).toUByteArray().asByteArray()
+}
+
+fun IntToByteArray(i: Int): ByteArray? {
```

Comment:
Looking at `array()` implementation, it seems it can't return an optional. It throws `UnsupportedOperationException` if backing is `null`. 

But it's not bad to keep it, just in case. We shouldn't rely on implementation details.

On the other side, you're returning non-optionals in the following methods that return `array()`. It's better to be consistent. Have either all returning optionals or all returning non optionals.